### PR TITLE
Hit reactions

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Ports TG's various hit reaction PRs and https://github.com/tgstation/tgstation/pull/13717 and  @KorPhaeron various reactive armor items.

Refactors `IsShield` to be based on `hit_reaction` instead.

This allows us to custom define a specific block chance for specific items instead of it being based under probalities for each `IsShield` proc.

It also allows us to more easily define custom effects when things wearing/holding something are hit.

Changes:
- Grenade have a chance to explode when the person holding them is shot
- Flamethrowers can release their tank's contents if the person holding them is shot
- Can no longer self-trigger shield checks
- Hugging someone/shaking them up bypasses shield checks (oh no, hugs of death)
- Riot shields have a +30% chance to block thrown projectiles
- Hunter pounces are now a thrown attack as far as shield go, and yes, can be blocked by riot shields now
- Energy shields no longer block anything other than beam and energy projectiles
- Being able to be pushed is based on the block chance of the item you're holding instead of it just being based on riot shields

FIxes
- Various armor penetration fixes, either not being respected or not being at the right location

Code Stuffs
- Throwing is now casted to items instead of objects. I can't think of a single non-item object you can get ahold of, and this allows thrown items to adhere to armor penetration
- NOSHIELD flag removed
- Simple Animals can now have armor penetration
- Makes the reactive armor's getting EMP'd *actually* update the wearer's mob worn sprite

Additions
- Adds in HoS's hardsuit
 - Not mapped, I only did this because I had to grab the sprites for it for another feature, and since it was a simple addition, might as well, again NOT ACCESSIBLE
- Implements @KorPhaeron shielded hardsuits
 - These are basically hardsuits that can absorb hits, negating the damage and such entirely--non-accessible, for now

:cl: Fox McCloud
add: Grenade have a chance to explode when the person holding them is shot
add: Flamethrowers can release their tank's contents if the person holding them is shot
Tweak: Can no longer self-trigger blocked shield events (ie: no self-harm to trigger a reactive teleport armor)
tweak: Hugging someone/shaking them up bypasses shield checks (oh no, hugs of death)
tweak: Riot shields have a +30% chance to block thrown projectiles
tweak: Hunter pounces are now a thrown attack as far as shield go, and yes, can be blocked by riot shields now
tweak: Energy shields no longer block anything other than beam and energy projectiles
tweak: Being able to be pushed is based on the block chance of the item you're holding instead of it just being based on riot shields
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/human/human_defense.dm [hunk 1]
```diff
@@ -315,60 +299,53 @@ emp_act
...
 				if(!istype(src,/mob/living/simple_animal/mouse))
-					msg_admin_attack("[key_name_admin(src)] was hit by a [O], thrown by [key_name_admin(M)]")
+					msg_admin_attack("[key_name_admin(src)] was hit by a [I], thrown by [key_name_admin(M)]")
 
 		//thrown weapon embedded object code.
-		if(dtype == BRUTE && istype(O,/obj/item))
-			var/obj/item/I = O
-			if(!I.is_robot_module())
+		if(dtype == BRUTE && istype(I))
+			if (!I.is_robot_module())
```
Comment (tigercat2000) [inline code/modules/mob/living/carbon/human/human_defense.dm]: no space between `if` and `(`

Comment (Chakirski) [inline code/modules/mob/living/carbon/human/human_defense.dm]: ayyyyy lmao

**Comment by Fox-McCloud** [comment]
DNM removed; it's been tested to the best that it's possible; things work, and are otherwise functional; fixed the sprite issue that was present.

**Comment by theColdflame** [comment]
A few questions:
-Does 'shot' include getting tased?
-Do the riot shield changes apply to the teleshield?
-Will hunters be stunned if their pounce is blocked?

**Comment by Fox-McCloud** [comment]
- Projectiles are projectiles, so it doesn't matter which type they are, if it's fired from a gun, it's a projectile, ergo "getting shot"
- Teleriot is still a riot shield, so yes
- No, but their pounce will be placed on cooldowd (they'll be "fatigued")

**Comment by KorPhaeron** [comment]
Tasers and disablers dont set off fuel tanks or grenades if you ported this without change

**Comment by KasparoVy** [comment]
God Bless :pray:

**Comment by Fox-McCloud** [comment]
whoops, almost Gave the RD the tele-armor that doesn't do anything.

**Pull request merged**